pkg/event_data_storage/accumulator.go

```diff
@@ -0,0 +1,265 @@
+package accumulator
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"sniffer/pkg/config"
+	"sniffer/pkg/ebpfeng"
+	evData "sniffer/pkg/ebpfev/v1"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+)
+
+const (
+	AccumulatorSize     = 10
+	DROP_EVENT_OCCURRED = "drop event occurred\n"
+)
+
+type containersEventStreamer struct {
+	streamDataChannelForContainerID map[string]chan evData.EventData
+	registerMutex                   sync.Mutex
+}
+
+type Accumulator struct {
+	data                            []map[string][]evData.EventData
+	startStatus                     bool
+	syncReaderWriterData            *sync.RWMutex
```

Comment:
I don't see often pointer to mutexes, this is normally used in specific cases - if so document why you need pointer here.